### PR TITLE
Adding Support For Onyx OS on Salt

### PR DESCRIPTION
**Pull request by anasbadaha**
Signed-off-by: Anas Badaha <anasb@mellanox.com>

### What does this PR do?
Adding Supports For Onyx OS in Salt

### What issues does this PR fix or reference?
Adding new features not bug fixes

### Tests written?

No

### Commits signed with GPG?

Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by rallytime** [comment]
Replaces #49690

File: salt/proxy/onyx.py [hunk 1]
```diff
@@ -0,0 +1,609 @@
...
+    password_line = user.split('\n')
+    password_line = password_line[1]
+    if not password_line:
+        return None
+    if '!!' in password_line:
+        return False
+    cur_hash = re.search(r'(\$[0-6](?:\$[^$ ]+)+)', str(password_line)).group(0)
+    if encrypted is False:
+        hash_type, cur_salt, hashed_pass = re.search(r'^\$([0-6])\$([^$]+)\$(.*)$', str(cur_hash)).groups()
+        new_hash = gen_hash(crypt_salt=cur_salt, password=password, algorithm=hash_algorithms[hash_type])
```
Comment (cachedout) [inline salt/proxy/onyx.py]: Be advised that this will raise an exception on Windows.

File: salt/proxy/onyx.py [hunk 2]
```diff
@@ -0,0 +1,609 @@
...
+            password='$5$2fWwO2vK$s7.Hr3YltMNHuhywQQ3nfOd.gAPHgs3SOBYYdGT3E.A' \\
+            encrypted=True
+    '''
+    password_line = get_user(username)
+    if encrypted is False:
+        if crypt_salt is None:
+            # onyx does not like non alphanumeric characters.  Using the random module from pycrypto
+            # can lead to having non alphanumeric characters in the salt for the hashed password.
+            crypt_salt = secure_password(8, use_random=False)
+        hashed_pass = gen_hash(crypt_salt=crypt_salt, password=password, algorithm=algorithm)
```
Comment (cachedout) [inline salt/proxy/onyx.py]: Same here, regarding an exception being raised on Windows.

**Comment by cachedout** [comment]
Hi @anasbadaha. Are you planning to come back to this? :)

**Comment by anasbadaha** [comment]
Hi @cachedout,

for sure I need to finish this! what is requested from me to finish it ? how can i fixed the failed checks ?

**Comment by cachedout** [comment]
@anasbadaha All that's needed for now is to look at the review comments I left on the PR. We need better handling for Windows machines. Once that is resolved, we can move forward. Thanks!

File: salt/proxy/onyx.py (hunk 1 above)
Comment (anasbadaha) [reply salt/proxy/onyx.py]: I have removed this function for now!

File: salt/proxy/onyx.py (hunk 2 above)
Comment (anasbadaha) [reply salt/proxy/onyx.py]: the same thing for this one as well!

**Comment by anasbadaha** [comment]
Hi @cachedout,

I have pushed a new commit without the two functions that will raise an exception on windows(set password, check password) we can work without them for now. can you please go over my code and review it again?

Best Regards,
Anas badaha

**Review by cachedout — APPROVED** [review]

**Comment by cachedout** [comment]
@anasbadaha It looks like this is failing some tests. Can you take a look?

**Comment by cachedout** [comment]
@cro Do you have a moment to look this over?

**Review by cro — APPROVED** [review]

**Comment by anasbadaha** [comment]
@cachedout what do you want me to do exactly ?

**Pull request merged**